run/opencl/tezos_kernel.cl

```diff
@@ -171,3 +176,21 @@ __kernel void pbkdf2_sha512_tezos_kernel(__global const pass_t *inbuffer,
 		state[idx].W[i] = tmp_out[i];
 	}
 }
+
+#undef z /* was defined by opencl_sha2.h and used in SHA256_ZEROS, but conflicts with the below */
+#include "ed25519-donna/ed25519-donna.c"
```

Comment:
Perhaps we should end `opencl_sha2.h` with `#undef z` and quite possibly more of such. That's a separate issue though.

Reply:
> Perhaps we should end `opencl_sha2.h` with `#undef z`

This would break uses of the `SHA*` macros from outside of that header file, and we have many of those. I think we should instead rename `z` and `Z` to something namespaced to that file, e.g. `SHA256_Z` and `SHA512_Z`. And yes, that change should be separate from this PR. I might do it along with my pending changes for #3525.